File: Lab3/Zadatak1/tiger.c
#include <stdlib.h>

typedef char const* (*PTRFUN)();

struct Tiger {
    PTRFUN* vtable;
    char const* name;
};

char const* name(void* this) {
    return ((struct Tiger*)this)->name;
}

char const* greet() {
    return "Pozdrav! Ja sam tigar";
}

char const* menu() {
    return "Macja hrana";
}

PTRFUN tigerVTable[3] = {
    (PTRFUN)name,
    (PTRFUN)greet,
    (PTRFUN)menu};

void constructTiger(struct Tiger* this, char const* name) {
    this->name = name;
    this->vtable = tigerVTable;
}

void* create(char const* name) {
    struct Tiger* this = malloc(sizeof(struct Tiger));
    constructTiger(this, name);
    return this;
}

File: Lab3/Zadatak1/Zadatak1.4/src/hr/fer/zemris/ooup/lab2/model/plugins/Tiger.java
package hr.fer.zemris.ooup.lab2.model.plugins;

import hr.fer.zemris.ooup.lab2.model.Animal;

public class Tiger extends Animal {

    private String animalName;

    public Tiger(String name) {
		this.animalName = name;
    }

    @Override
    public String name() {
        return animalName;
    }

    @Override
    public String greet() {
        return "Pozdrav! Ja sam tigar.";
    }

    @Override
    public String menu() {
        return "Macja hrana.";
    }
}